src/Controller/CRUDController.php

```diff
@@ -57,16 +70,115 @@ class CRUDController extends Controller
      */
     protected $admin;
 
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    /**
+     * @var Pool
+     */
+    private $pool;
+
     /**
      * The template registry of the related Admin class.
      *
      * @var TemplateRegistryInterface
      */
     private $templateRegistry;
 
-    public function setContainer(?ContainerInterface $container = null): void
-    {
-        $this->container = $container;
+    /**
+     * @var BreadcrumbsBuilder
+     */
+    private $breadcrumbsBuilder;
+
+    /**
+     * @var Environment
+     */
+    private $twig;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var SessionInterface
+     */
+    private $session;
+
+    /**
+     * @var KernelInterface
+     */
+    private $kernel;
+
+    /**
+     * @var AuditManagerInterface
+     */
+    private $auditManager;
+
+    /**
+     * @var Exporter|null
+     */
+    private $exporter;
+
+    /**
+     * @var AdminExporter|null
+     */
+    private $adminExporter;
+
+    /**
+     * @var CsrfTokenManagerInterface|null
+     */
+    private $csrfTokenManager;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var AdminObjectAclManipulator|null
+     */
+    private $adminObjectAclManipulator;
+
+    /**
+     * @var AdminObjectAclUserManager|null
+     */
+    private $adminObjectAclUserManager;
+
+    public function __construct(
+        RequestStack $requestStack,
+        Pool $pool,
+        TemplateRegistryInterface $templateRegistry,
+        BreadcrumbsBuilder $breadcrumbsBuilder,
+        Environment $twig,
+        TranslatorInterface $translator,
+        SessionInterface $session,
+        KernelInterface $kernel,
+        AuditManagerInterface $auditManager,
+        ?Exporter $exporter = null,
+        ?AdminExporter $adminExporter = null,
+        ?CsrfTokenManagerInterface $csrfTokenManager = null,
+        ?LoggerInterface $logger = null,
+        ?AdminObjectAclManipulator $adminObjectAclManipulator = null,
+        ?AdminObjectAclUserManager $adminObjectAclUserManager = null
+    ) {
+        $this->requestStack = $requestStack;
+        $this->pool = $pool;
+        $this->templateRegistry = $templateRegistry;
+        $this->breadcrumbsBuilder = $breadcrumbsBuilder;
+        $this->twig = $twig;
+        $this->translator = $translator;
+        $this->session = $session;
+        $this->kernel = $kernel;
+        $this->auditManager = $auditManager;
+        $this->exporter = $exporter;
+        $this->adminExporter = $adminExporter;
```

Comment:
why 2 exporters?

Reply:
One for sonata.admin.admin_exporter and other for sonata.exporter.exporter

Reply:
I dont see usages of sonata.exporter.exporter on the previous code, can you point me on them?

Reply:
https://github.com/sonata-project/SonataAdminBundle/blob/3.x/src/Controller/CRUDController.php#L898